app/main.py

```diff
@@ -1,23 +1,39 @@
 import init_django_orm  # noqa: F401
 
-from django.db.models import QuerySet
+from django.db.models import QuerySet, Q, F
+
+from db.models import LoyaltyProgram, Customer, LoyaltyProgramParticipant
 
 
 def all_loyalty_program_names() -> QuerySet:
-    pass
+    return LoyaltyProgram.objects.values_list(
+        "name", "bonus_percentage"
+    )
 
 
 def not_active_customers() -> QuerySet:
-    pass
+    return LoyaltyProgramParticipant.objects.exclude(
+        last_activity__gte="2022-01-01").filter(
+        last_activity__gte="2021-01-01").values(
+        "customer__first_name"
+    )
 
 
 def most_active_customers() -> QuerySet:
-    pass
+    return LoyaltyProgramParticipant.objects.order_by(
+        -F("sum_of_spent_money")).values_list(
```

Comment:
I don't think `F` is needed here. Can't you just add a minus to the beginning of the string?

Reply:
Yes, it's a good comment. I'll implement it.